src/hotspot/cpu/x86/stubGenerator_x86_64.cpp

```diff
@@ -4326,10 +4326,14 @@ void StubGenerator::generate_compiler_stubs() {
     if (libsimdsort != nullptr) {
       log_info(library)("Loaded library %s, handle " INTPTR_FORMAT, JNI_LIB_PREFIX "simdsort" JNI_LIB_SUFFIX, p2i(libsimdsort));
 
-      snprintf(ebuf_, sizeof(ebuf_), VM_Version::supports_avx512dq() ? "avx512_sort" : "avx2_sort");
+      snprintf(ebuf_, sizeof(ebuf_),
+               ((VM_Version::is_intel() || (VM_Version::is_amd() && (VM_Version::cpu_family() > 0x19)))
+                && VM_Version::supports_avx512dq()) ? "avx512_sort" : "avx2_sort");
```

Comment:
Perhaps factor the expression to a separate method rather than repeat it three times?

Can we add some constant with a descriptive name for the CPU family rather than directly using 0x19?